File: js/node/factory/pizzaIngredientFactory/NYPizzaIngredientFactory.js
require('./PizzaIngredientFactory');

require('../dough/ThinCrustDough');
require('../sauce/MarinaraSauce');
require('../cheese/ReggianoCheese');
require('../veggies/Garlic');
require('../veggies/Onion');
require('../veggies/Mushroom');
require('../veggies/RedPepper');
require('../pepperoni/SlicedPepperoni');
require('../clams/FreshClams');

NYPizzaIngredientFactory = function() {
    PizzaIngredientFactory.apply(this);
}
NYPizzaIngredientFactory.prototype = new PizzaIngredientFactory();
NYPizzaIngredientFactory.prototype.createDough = function() {
    return new ThinCrustDough();
}
NYPizzaIngredientFactory.prototype.createSauce = function() {
    return new MarinaraSauce();
}
NYPizzaIngredientFactory.prototype.createCheese = function() {
    return new ReggianoCheese();
}
NYPizzaIngredientFactory.prototype.createVeggies = function() {
    this.veggies = [];
    this.veggies.push(new Garlic());
    this.veggies.push(new Onion());
    this.veggies.push(new Mushroom());
    this.veggies.push(new RedPepper());
    return this.veggies;
}
NYPizzaIngredientFactory.prototype.createPepperoni = function() {
    return new SlicedPepperoni();
}
NYPizzaIngredientFactory.prototype.createClam = function() {
    return new FreshClams();
}

File: java/1.7/factory/src/com/pizzaIngredientFactory/NYPizzaIngredientFactory.java
package com.pizzaIngredientFactory;

import com.cheese.Cheese;
import com.cheese.ReggianoCheese;
import com.clams.Clams;
import com.clams.FreshClams;
import com.dough.Dough;
import com.dough.ThinCrustDough;
import com.pepperoni.Pepperoni;
import com.pepperoni.SlicedPepperoni;
import com.sauce.MarinaraSauce;
import com.sauce.Sauce;
import com.veggies.Garlic;
import com.veggies.Mushroom;
import com.veggies.Onion;
import com.veggies.RedPepper;
import com.veggies.Veggies;

public class NYPizzaIngredientFactory implements PizzaIngredientFactory {
 
	public Dough createDough() {
		return new ThinCrustDough();
	}
 
	public Sauce createSauce() {
		return new MarinaraSauce();
	}
 
	public Cheese createCheese() {
		return new ReggianoCheese();
	}
 
	public Veggies[] createVeggies() {
		Veggies veggies[] = { new Garlic(), new Onion(), new Mushroom(), new RedPepper() };
		return veggies;
	}
 
	public Pepperoni createPepperoni() {
		return new SlicedPepperoni();
	}

	public Clams createClam() {
		return new FreshClams();
	}
}
